visualization/include/pcl/visualization/impl/pcl_visualizer.hpp

```diff
@@ -1701,8 +1701,32 @@ pcl::visualization::PCLVisualizer::addPolygonMesh (
   {
     // Create polys from polyMesh.polygons
     vtkSmartPointer<vtkCellArray> cell_array = vtkSmartPointer<vtkCellArray>::New ();
-    vtkIdType *cell = cell_array->WritePointer (vertices.size (), vertices.size () * (max_size_of_polygon + 1));
     int idx = 0;
+
+#ifdef VTK_CELL_ARRAY_V2
+    if (!lookup.empty ())
+    {
+      for (std::size_t i = 0; i < vertices.size (); ++i, ++idx)
```

Comment:
range-for loop (and remove the comments in the #else section)

Reply:
Just realized, idx is being modified, so range-for would not be possible without something like:
```cpp
const auto idx = vertices.size() + std::accumulate(vectices.begin(), vertices.end(), 0,
    [](const auto& sum, const auto& vertex) { return sum + vertex.vertices.size(); });
```